code/modules/mob/inventory.dm

```diff
@@ -26,10 +26,9 @@
 
 
 /mob/proc/get_item_for_held_index(i)
-	if(i > 0 && i <= held_items.len)
-		return held_items[i]
-	return FALSE
-
+    if(!i || !held_items.len)
+        return null
+    return held_items[i]
```

Comment:
This should return null if i is outside the bounds of the array instead of runtiming

Reply:
The behaviour is inconsistent then, it returns null on 0 or if there are no held items but will runtime if i is any non-zero number outside the bounds of a non-zero array.

Reply:
The problem is that this is called on many mobs that don't even have hands at all, which results in many runtimes if the zero is not handled.

Reply:
Only problem I can see is if something has an odd number of hands and we try to find their inactive hand index, since for 3 it'll look for a non-existent hand at 4, which then the logic for that actually returns zero since 4 is outside of the list length (see `mob/proc/get_inactive_hand_index()` in the same file.)

Reply:
If some code outside of this function doesn't work then its their problem, this function should do exactly as it describes and act in a consistent manner.
Either it should always runtime when out of bounds, or it should always return null when out of bounds.